app/jobs/tts_job.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# This jobs generates TTS audio attachments for all words with enabled TTS but without audio attachment.
+# It iterates automatically through all words and ends, when there are no more words to process.
+# There should be a cronjob running this job every 5 minutes and avoid duplicates, to make sure, that the jobs
+# is always running.
+class TtsJob < ApplicationJob
+  queue_as :default
+
+  # There is a rate limit for the TTS API. We shouldn't hit that limit even without pause, but to be
+  # absolute sure, we wait a second after each request.
+  PAUSE = 1.second
+
+  def perform
+    @logger = Logger.new(Rails.root.join("log", "tts.log"))
+    @logger.info "Starting TTS job"
+
+    loop do
+      word = find_next_word
+
+      unless word
+        @logger.info "Found no more words to process. Exiting."
+        return true
+      end
+
+      generate_audio word
+      sleep PAUSE
+    end
+  end
+
+  private def generate_audio(word)
+    @logger.info "Processing word #{word.id} (#{word.name})"
+
+    word.audio.attach(
+      io: TtsGenerator.call(word.name),
```

Comment:
If Google throws an error (e.g. service not reachable, or maybe the word can't be spoken for whatever reason), the job would crash, right? It then would be started again by cron, potentially choosing the same word again (as long as the SQL query result is stable). If the issue is with the word, we would fail continuously.

Would it make sense to catch possible exceptions here, maybe log something and then continue?

Reply:
Fixed.